tests/tensorflow_autolog/test_tensorflow2_autolog.py

```diff
@@ -285,6 +287,47 @@ def test_tf_keras_autolog_early_stop_logs(tf_keras_random_data_run_with_callback
     assert history.history["loss"][restored_epoch] == metric_history[-1].value
 
 
+@pytest.mark.large
+@pytest.mark.parametrize("restore_weights", [True])
+@pytest.mark.parametrize("callback", ["early"])
+@pytest.mark.parametrize("patience", [0, 1, 5])
+def test_tf_keras_autolog_batch_metrics_logger_logs_expected_metrics(
```

Comment:
Is there a non-tf-keras test case that we should add? Vanilla TensorFlow workflows are also important here.

Reply:
No. We only add batch metrics logger to tf.keras `fit` and `fit_generator`.  We don't use the logger for Vanilla TensorFlow. For `tf.estimator` we use TensorBoard to log metrics, and  that already has a custom batching solution implemented `_metric_queue`. Correct me if I am wrong

Reply:
Right! We had this discussion previously. Thanks for the reminder!